Automattic/jetpack
# Publicize: point wp-admin users to WP.com calypso sharing page instead

This PR seeks to remove the publicize connection settings from WP.com's wp-admin and instead point users to the Calypso sharing page for their site.

![screenshot from 2015-12-18 14-55-01](https://cloud.githubusercontent.com/assets/260253/11909090/5b1bf2a4-a597-11e5-9059-4ffbc954f844.png)

The code that powers the wp-admin sharing screen is considered legacy code, and is out of date with our latest Publicize connections codebase (powered by the WP.com REST API). We're taking aim at cleaning up old, less reliable code and reducing our technical debt, while offering the best user experience for users.

The XML-RPC methods will continue to be supported for the foreseeable future, even though the UI will be gone, but may be retired at some point once usage of older versions of Jetpack drops to low enough numbers.

With that in mind, it would be possible to re-create a wp-admin experience using the REST API. All of the endpoints are fully available and documented here: https://developer.wordpress.com/docs/api/#sharing

This PR is a merge of r128505-wpcom with additional removals for some JP-specific code.

To test:

* Apply PR to your Jetpack site
* Activate Publicize
* Verify that you get the screenshot above in place of the publicize settings
* Verify that the button and link take you to the expected Calypso Sharing page
* Verify that you can still add and remove publicize connections
* Verify that your added connections show up in wp-admin's Editor and Calypso Editor's as available connections
* Verify that posting a post publicizes the post as expected
* Verify that all of the above occurs without any php errors in your site's error log
* Login with a secondary user on your JP site, verify that you're still presented with this:

![screenshot from 2015-12-18 14-54-46](https://cloud.githubusercontent.com/assets/260253/11909166/0c5bad84-a598-11e5-82a3-521b18c5dd73.png)
 

cc @timmyc @aduth @nylen for review
